Libraries/NativeModules/specs/NativeDialogManagerAndroid.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2004-present Facebook. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+
+import Platform from '../../Utilities/Platform';
+
+type DialogAction = 'buttonClicked' | 'dismissed';
+type DialogButtonKey = 'buttonPositive' | 'buttonNegative' | 'buttonNeutral';
+type DialogOptions = {|
+  +title?: string,
+  +message?: string,
+  +buttonPositive?: string,
+  +buttonNegative?: string,
+  +buttonNeutral?: string,
+  +items?: Array<string>,
+  +cancelable?: boolean,
+|};
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {|
+    buttonClicked: DialogAction,
+    dismissed: DialogAction,
+    buttonPositive: DialogButtonKey,
+    buttonNegative: DialogButtonKey,
+    buttonNeutral: DialogButtonKey,
+  |};
+  +showAlert: (
+    config: DialogOptions,
+    onError: (string) => void,
+    onAction: (action: DialogAction, buttonKey: DialogButtonKey) => void,
```

Comment:
Looking through the code, it seems like `onAction` can also be invoked without the `buttonKey`:

```Java
    @Override
    public void onDismiss(DialogInterface dialog) {
      if (!mCallbackConsumed) {
        if (getReactApplicationContext().hasActiveCatalystInstance()) {
          mCallback.invoke(ACTION_DISMISSED);
          mCallbackConsumed = true;
        }
      }
    }
```

So we should make this argument nullable.